ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1841,6 +1908,60 @@ void addChunkId(StoreKey storeKey, long chunkSize, int chunkIndex) {
       indexToChunkIdsAndChunkSizes.put(chunkIndex, new Pair<>(storeKey, chunkSize));
     }
 
+    /**
+     * Validate the reserved metadata chunk id is passed in correctly where it needs to be specified.
+     * @throws RouterException in case of any errors.
+     */
+    private void validateReservedMetadataChunkId() throws RouterException {
+      if (!options.isChunkUpload() && Objects.nonNull(passedInBlobProperties.getReservedMetadataBlobId())) {
+        reservedMetadataIdMetrics.reservedMetadataPassedInForNonChunkedUploadCount.inc();
+        throw new RouterException("Reserved metadata should be passed in for request that isn't chunked upload.",
```

Comment:
Is this error message correct? I think you meant "should not". Regardless, can you simplify your messages? Something like, "Only chunk uploads can reserve metadata blob id"

Reply:
done.